src/pages/TrueSightV2/components/SearchWithDropDown.tsx

```diff
@@ -0,0 +1,542 @@
+        </Text>
+      </td>
+    </DropdownItem>
+  )
+}
+
+const SearchResultTableWrapper = ({ header, children }: { header?: ReactNode; children?: ReactNode }) => {
+  return (
+    <DropdownSection>
+      <colgroup>
+        <col style={{ width: '200px', minWidth: 'fit-content' }} />
+        <col style={{ width: '100px', minWidth: 'auto' }} />
+        <col style={{ width: '100px' }} />
+        <col style={{ width: '60px' }} />
+      </colgroup>
+      <thead>
+        <tr>
+          <th>{header}</th>
+          <th style={{ textAlign: 'left' }}>KyberScore</th>
+          <th style={{ textAlign: 'left' }}>Price</th>
+          <th style={{ textAlign: 'right' }}>24H</th>
+        </tr>
+      </thead>
+      <tbody>{children}</tbody>
+    </DropdownSection>
+  )
+}
+
+const SearchWithDropdown = () => {
+  const theme = useTheme()
+  const { mixpanelHandler } = useMixpanel()
+  const [searchParams] = useSearchParams()
+  const { pathname } = useLocation()
+
+  const [expanded, setExpanded] = useState(false)
+  const [search, setSearch] = useState('')
+  const [height, setHeight] = useState(0)
+  const wrapperRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const dropdownRef = useRef<HTMLDivElement>(null)
+  const contentRef = useRef<HTMLDivElement>(null)
+
+  const debouncedSearch = useDebounce(search, 1000)
+  const { data: searchResult, isFetching } = useSearchTokenQuery(
+    { q: debouncedSearch, size: 10 },
+    { skip: debouncedSearch === '' },
+  )
+  const [history, setHistory] = useLocalStorage<Array<ITokenSearchResult>>('kyberai-search-history')
+  const saveToHistory = (token: ITokenSearchResult) => {
+    if (!(history && history.findIndex(t => t.address === token.address && t.chain === token.chain) >= 0)) {
+      setHistory([token, ...(history || [])].slice(0, 3))
+    }
+  }
+
+  const listType = (searchParams.get('listType') as KyberAIListType) || KyberAIListType.BULLISH
+
+  const { data: top5bullish, isLoading: isBullishLoading } = useTokenListQuery({
+    type: KyberAIListType.BULLISH,
+    page: 1,
+    pageSize: 5,
+  })
+
+  const { data: top5bearish, isLoading: isBearishLoading } = useTokenListQuery({
+    type: KyberAIListType.BEARISH,
+    page: 1,
+    pageSize: 5,
+  })
+
+  const haveSearchResult = debouncedSearch !== '' && searchResult && searchResult.length > 0 && !isFetching
+  const noSearchResult = debouncedSearch !== '' && searchResult && searchResult.length === 0 && !isFetching
+  const isLoading = isFetching && search === debouncedSearch
+  const above768 = useMedia(`(min-width:${MEDIA_WIDTHS.upToSmall}px)`)
+
+  useOnClickOutside(wrapperRef, () => setExpanded(false))
+  useEffect(() => {
+    if (!inputRef.current) return
+    const inputEl = inputRef.current
+    const onFocus = () => {
+      setExpanded(true)
+    }
+    inputEl.addEventListener('focusin', onFocus)
+    return () => {
+      inputEl.removeEventListener('focusin', onFocus)
+    }
+  }, [])

```

Comment:
This effect likely to be useless since it would be run only once at the first time the component renders. But we got the condition to check `inputRef.current` and halt the execution.

Reply:
only a safety check, most of the time the inputRef will available. I think it's fine to use it like that. This event listener is important for the component.